source/filter.js

```diff
@@ -0,0 +1,37 @@
+"use strict";
+
+const map = {
+  "&": "&amp;",
+  "<": "&lt;",
+  ">": "&gt;",
+  '"': "&quot;",
+  "'": "&#39;",
+};
+
+
+/**
+ * Экранирует символы в html, оставляя только разрешенные теги
+ * @param {string} text - фильтруемый текст
+ * @param {Array<string>} tags - список разрешенных тегов
+ * @returns {string} отфильтрованный текст
+ */
+const filter = (text, tags = []) => {
+  if (!(typeof text === "string")) {
+    throw new Error("expected String");
+  }
+  if (!Array.isArray(tags) || tags.some((el) => typeof el !== "string")) {
+    throw new Error("expected Array of strings");
+  }
+  const resArr = text.split(/([&"<>'])/g);
```

Comment:
```suggestion
  const resArr = text.split(/([&"<>'])/);
```
для `split` не влияет регулярка с `global` флаг или без него